src/Razor/test/Microsoft.AspNetCore.Razor.VSCode.FunctionalTest/tests/Rename.test.ts

```diff
@@ -0,0 +1,139 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as assert from 'assert';
+import { afterEach, before, beforeEach } from 'mocha';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import {
+    mvcWithComponentsRoot,
+    pollUntil,
+    waitForProjectReady,
+} from './TestUtil';
+
+let csPath: string;
+let cshtmlPath: string;
+let razorPath: string;
+
+suite('Rename', () => {
+    before(async () => {
+        await waitForProjectReady(mvcWithComponentsRoot);
+    });
+
+    beforeEach(async () => {
+        razorPath = path.join(mvcWithComponentsRoot, 'Views', 'Shared', 'NavMenu.razor');
+        cshtmlPath = path.join(mvcWithComponentsRoot, 'Views', 'Home', 'Index.cshtml');
+        csPath = path.join(mvcWithComponentsRoot, 'Test.cs');
+    });
+
+    afterEach(async () => {
+        await vscode.commands.executeCommand('workbench.action.revertAndCloseActiveEditor');
+        await pollUntil(() => vscode.window.visibleTextEditors.length === 0, 1000);
+    });
+
+    test('Can rename symbol within .razor', async () => {
+        const razorDoc = await vscode.workspace.openTextDocument(razorPath);
+        const razorEditor = await vscode.window.showTextDocument(razorDoc);
+        const expectedNewText = 'World';
+        const firstLine = new vscode.Position(0, 0);
+        await razorEditor.edit(edit => edit.insert(firstLine, '@hello\n'));
+        await razorEditor.edit(edit => edit.insert(firstLine, '@{ var hello = "Hello"; }\n'));
+
+        await new Promise(r => setTimeout(r, 3000));
```

Comment:
Ideally these should be replaced with something that waits for O# to fully initialize our documents.